sklearn/utils/tests/test_validation.py

```diff
@@ -835,3 +836,63 @@ def test_check_scalar_invalid(x, target_name, target_type, min_val, max_val,
                      min_val=min_val, max_val=max_val)
     assert str(raised_error.value) == str(err_msg)
     assert type(raised_error.value) == type(err_msg)
+
+
+psd_cases = [('nominal', (1., 2.), np.array([1., 2.]), None, ""),
+             ('insignificant_imag', (5., 5e-5j), np.array([5., 0.]), None, ""),
+             ('significant neg', (5., -1.), np.array([5., 0.]),
+              PSDSpectrumWarning, "There are significant negative eigenval"),
+             ('insignificant neg', (5, -5e-5), np.array([5., 0.]), None, ""),]
+
+
+@pytest.mark.parametrize("input, output, w_type, w_msg",
+                         [c[1:] for c in psd_cases],
+                         ids=[c[0] for c in psd_cases])
+@pytest.mark.parametrize("warn_on_zeros", [True, False])
+def test_check_psd_eigenvalues_valid(input, output, w_type, w_msg,
+                                     warn_on_zeros):
+    """Test that check_psd_eigenvalues returns the right output for valid
+    input, possibly raising the right warning"""
+
+    with pytest.warns(w_type, match=w_msg) as w:
+        assert_array_equal(
+            check_psd_eigenvalues(input, warn_on_zeros=warn_on_zeros), output)
+    if w_type is None:
+        assert not w
+
+
+def test_check_psd_eigenvalues_bad_conditioning_warning():
+    """Test that check_psd_eigenvalues raises a warning for bad conditioning
+    when warn_on_zeros is set to True, and does not when it is set to False"""
+
+    input = (5, 4e-12)
+    output = np.array([5., 0.])
```

Comment:
Here and everywhere else, you shouldn't need to explicitly set floats (`5.` -> `5`)

Reply:
ok